mobile-ui/src/main/java/org/sagebionetworks/research/mobile_ui/recorder/distance/DistanceJsonAdapter.java

```diff
@@ -0,0 +1,156 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.sagebionetworks.research.mobile_ui.recorder.distance;
+
+import android.icu.text.SimpleDateFormat;
+import android.location.Location;
+import android.os.Build;
+import android.os.Build.VERSION;
+import android.os.Build.VERSION_CODES;
+import android.os.SystemClock;
+import android.support.annotation.NonNull;
+
+import com.google.gson.JsonObject;
+
+import org.sagebionetworks.research.mobile_ui.recorder.data.FormatHelper;
+
+import java.util.Locale;
+
+/**
+ * Helper to convert Location's measured by a DistanceRecorder into JsonObjects for writing to a file.
+ */
+public class DistanceJsonAdapter {
+    private static final String SHARED_PREFS_KEY = "LocationRecorder";
+    private static final String LAST_RECORDED_DIST_KEY = "LastRecordedTotalDistance";
+    public static final String COORDINATE_KEY  = "coordinate";
+    public static final String LONGITUDE_KEY   = "longitude";
+    public static final String LATITUDE_KEY    = "latitude";
+    public static final String ALTITUDE_KEY    = "altitude";
+    public static final String ACCURACY_KEY    = "accuracy";
+    public static final String COURSE_KEY      = "course";
+    public static final String RELATIVE_LATITUDE_KEY = "relativeLatitude";
+    public static final String RELATIVE_LONGITUDE_KEY = "relativeLongitude";
+    public static final String SPEED_KEY       = "speed";
+    public static final String TIMESTAMP_DATE_KEY = "timestampDate";
+    public static final String TIMESTAMP_IN_SECONDS_KEY = "timestamp";
+    public static final String UPTIME_IN_SECONDS_KEY = "uptime";
+
+    /**
+     * Returns a JsonObject that represents the given location.
+     * @param location The location to represent in Json.
+     * @param usesRelativeCoordinates `true` if coordinates relative to the first location should be used.
+     *                               `false` if absolute longitude and latitude should be used.
+     * @param firstLocation The first location this distance recorder measured. Even if absolute coordinates are being
+     *                      used this shouldn't be null as the timestamp is also used.
+     * @return a JsonObject that represents the given location.
+     */
+    @NonNull
+    public static JsonObject getJsonObject(@NonNull Location location,
```

Comment:
Do we need to use JsonObject? Or can we map to another POJO and use Gson?

Reply:
I will implement it that way.